### PR TITLE
Free daily market

### PR DESCRIPTION
**Pull request by IanPhilips**
Users get a free daily market and confetti when they create new markets! The confetti works for ALL owned, newly created markets - not just the free ones.

### CODE REVIEW

File: web/lib/firebase/contracts.ts [hunk 1]
```diff
@@ -121,6 +121,19 @@ export async function listAllContracts(): Promise<Contract[]> {
   return snapshot.docs.map((doc) => doc.data() as Contract)
 }
 
+export async function userHasCreatedContractToday(
+  userId: string
+): Promise<boolean> {
+  // uses utc time like the server
+  const todayAtMidnight = dayjs.utc().startOf('day').valueOf()
+  return listContracts(userId).then((contracts) => {
```
Comment (jahooma) [inline web/lib/firebase/contracts.ts:129]: You could switch this to async/await style, e.g.
```
const contracts = await listContracts(userId)
...
```

File: web/lib/firebase/contracts.ts [hunk 2]
```diff
@@ -121,6 +121,19 @@ export async function listAllContracts(): Promise<Contract[]> {
   return snapshot.docs.map((doc) => doc.data() as Contract)
 }
 
+export async function userHasCreatedContractToday(
+  userId: string
+): Promise<boolean> {
+  // uses utc time like the server
```
Comment (jahooma) [inline web/lib/firebase/contracts.ts:127]: Nice attention to detail!

File: web/pages/[username]/[contractSlug].tsx [hunk 3]
```diff
@@ -119,6 +122,17 @@ export function ContractPageContent(props: FirstArgument<typeof ContractPage>) {
 
   return (
     <Page rightSidebar={rightSidebar}>
+      {user &&
+        contract.creatorId === user.id &&
+        Date.now() - contract.createdTime < 10 * 1000 && (
```
Comment (jahooma) [inline web/pages/[username]/[contractSlug].tsx:127]: After react renders, it won't necessarily know to render again when this condition changes. A more robust alternative would be to set a timeout that calls a set state. E.g. https://usehooks-ts.com/react-hook/use-timeout

File: web/lib/firebase/contracts.ts (hunk 2 above)
Comment (jahooma) [reply web/lib/firebase/contracts.ts:127]: A very small style nit is to use capital letters at the start of a comment. At Google we even ended with a period, though I wouldn't enforce that.

**Review by jahooma — CHANGES_REQUESTED** [review]
Awesome! This looks pretty good. Just a few comments.

File: web/pages/[username]/[contractSlug].tsx (hunk 3 above)
Comment (IanPhilips) [reply web/pages/[username]/[contractSlug].tsx:127]: Hmmm the logic should check against a separate time, not the time since it loaded.

**Review by jahooma — APPROVED** [review]
Awesome, looks good. Thanks Ian!

**Pull request merged**